torch/csrc/jit/import.cpp

```diff
@@ -138,6 +143,185 @@ void ScriptModuleDeserializer::loadTensorTable(torch::ModelDef* model_def) {
   }
 }
 
+struct Unpickler {
+  Unpickler(
+      void* data,
+      uint64_t size,
+      const std::vector<at::Tensor>& tensor_table)
+      : bytes_(static_cast<const uint8_t*>(data)),
+        end_ptr_(bytes_ + size),
+        tensor_table_(tensor_table) {}
+
+  const std::vector<IValue>& get_ivalue_list() {
+    run();
+    AT_ASSERT(stack_.size() == 1);
+    return stack_[0].toGenericListRef();
+  }
+
+ private:
+  uint8_t readByte() {
+    uint8_t val = *bytes_;
```

Comment:
These should probably guard not reading off the end of a improperly written input.